### DISCUSSION TITLE
Dnn module DOM Depth

### DISCUSSION THREAD

**Discussion by thienvc** [dnnsoftware/Dnn.Platform#6266] [General]
I'm trying to reduce the depth of the DOM in DNN, for example the most commonly used basic module is HTML:
Default xpath to content of html module:
**Xpath**: `/html/body/form/main/div[2]/div[2]/div[2]/div/div[2]`
![image](https://github.com/user-attachments/assets/2263f42b-c4d8-48f3-b7f6-e67a17c747ec)

After modified:
**Xpath**: `/html/body/form/main/div[2]/div[2]`
![image](https://github.com/user-attachments/assets/f46c1e01-e004-418c-8ec7-a340588266bb)

reducing the depth of Dom makes the web run faster, lighter, browsers render faster, Search Engines also like it more

**What I want to ask is, is there anything similar to what I want in DNN?
Is there a more general solution in DNN?**

**Comment by mitchelsellers** [comment]
It looks like you are in edit mode in the first entry, based on some of the items that you see.  There isn't really any way to "trim" the structure except for theme/container related items to try and optimize.

**Reply by thienvc** [reply]
Yes, I opened 2 versions for testing, my code excludes the situation of being logged in and having Edit module permission
When having edit module permission, everything stays the same, I just try to clean up the DOM when in view-only mode
